common/src/warp_runner/manager/commands/multipass_commands.rs

```diff
@@ -118,9 +121,30 @@ pub async fn handle_multipass_cmd(cmd: MultiPassCmd, warp: &mut super::super::Wa
         MultiPassCmd::CreateIdentity { .. } | MultiPassCmd::TryLogIn { .. } => {
             // do nothing and drop the rsp channel
         }
-        MultiPassCmd::RequestFriend { did, rsp } => {
-            let r = warp.multipass.send_request(&did).await;
-            let _ = rsp.send(r);
+        MultiPassCmd::RequestFriend { id, rsp } => {
+            // First attempt using a did
+            if let Ok(some) = DID::from_str(id.as_str()) {
```

Comment:
You could do a match condition to try for the did and if it errors to then proceed to attempt to fetch the identity by the username, grab the did get and then pass it along to `MultiPass::send_request`

eg

```rust
let did = match DID::from_str(&id) {
  Ok(did) => did,
  _ => {
    match warp.multipass.get_identity(Identifier::Username(id)).await.ok().and_then(list| list.first().cloned().map(|i|  i.did_key()) {
      Some(did) => did,
      _ => { 
          let _ = rsp.send(Err(Error::IdentityInvalid));
          return;
      }
    }
  }
}

let r = warp.multipass.send_request(&did).await;
let _ = rsp.send(r);
return;
```